src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -253,6 +268,26 @@ export class SbbDatepicker implements ComponentInterface {
     event.target.value = match?.[0] ?? null;
   }
 
+  private _getValidValue(value: string): string {
+    if (!value) {
+      return '';
+    }
+
+    const match: RegExpMatchArray = value.match(FORMAT_DATE);
+
+    if (match && match?.index === 0) {
+      return this._formatValue(value);
+    }
+
+    if (!Number.isNaN(+value) && Number.isInteger(+value)) {
```

Comment:
I think this check is enough as NaN is not an Integer?
```suggestion
    if (Number.isInteger(+value)) {
```